odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -1294,6 +1295,214 @@ def test_properties_field_definition_update(self):
         self.assertEqual(expected_properties, sql_properties)
 
     @mute_logger('odoo.fields')
+    def test_properties_field_search(self):
+        self.env['test_new_api.message'].search(
+            [('id', 'not in', (self.message_1 | self.message_2 | self.message_3).ids)]
+        ).unlink()
+
+        # search on text properties
+        self.message_1.attributes = [{
+            'name': 'mychar',
+            'type': 'char',
+            'value': 'Test',
+            'definition_changed': True,
+        }]
+        self.message_2.attributes = {'mychar': 'TeSt'}
+
+        messages = self.env['test_new_api.message'].search([('attributes.mychar', '=', 'Test')])
+        self.assertEqual(messages, self.message_1, "Should be able to search on a properties field")
+        messages = self.env['test_new_api.message'].search([('attributes.mychar', '=', '"Test"')])
+        self.assertFalse(messages)
+        messages = self.env['test_new_api.message'].search([('attributes.mychar', 'ilike', 'test')])
+        self.assertEqual(messages, self.message_1 | self.message_2)
+        messages = self.env['test_new_api.message'].search([('attributes.mychar', 'not ilike', 'test')])
+        self.assertFalse(messages)
+        messages = self.env['test_new_api.message'].search([('attributes.mychar', 'ilike', '"test"')])
+        self.assertFalse(messages)
+
+        for search in ['mychar!', 'my!char', '!mychar']:
```

Comment:
Should you test with other special characters? Like some that would allow using SQL injections?